src/test/java/org/example/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,165 @@
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import org.example.dto.CategoryDto;
+import org.example.service.CategoryService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private CategoryService categoryService;
+
+    private CategoryDto categoryDto1;
+    private CategoryDto categoryDto2;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders.webAppContextSetup(applicationContext)
+                          .apply(SecurityMockMvcConfigurers.springSecurity())
+                          .build();
+    }
+
+    @BeforeEach
+    void setUp() {
+        categoryDto1 = new CategoryDto();
+        categoryDto1.setName("First category");
+        categoryDto1.setDescription("Description of first category");
+
+        categoryDto2 = new CategoryDto();
+        categoryDto2.setName("Second category");
+        categoryDto2.setDescription("Description of second category");
+    }
+
+    @Test
+    @DisplayName("Create a new category")
+    @Sql(scripts = {"classpath:db/category-controller-tests/"
+                    + "after/delete-categories-from-categories-table.sql"},
+                executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
```

Comment:
if you have the same setup in each method, it would make sense to move it to the class level instead, fix everywhere